### PR TITLE
Feature/NP-1601 Review v4 build of DS

### PR DESCRIPTION
**Pull request by celonicolosi**
- all scenarios reviewed
- all falling / future releases / NP-xxxx tags removed
- some scenarios fixed
- one scenario removed (Scenario: Content is not auto-opened if previously interacted with)

### CODE REVIEW

File: testing/features/support/components/header/standard.rb [hunk 1]
```diff
@@ -8,10 +8,10 @@ class Standard < ::Base
     include Helpers::Javascript
 
     element :change_language, "a", text: "Cymraeg"
-    element :login, "[data-testid='account-link']"
```
Comment (luke-hill) [inline testing/features/support/components/header/standard.rb:11]: We probably don't want to undo data id hooking if possible. Also the data id probably still exists.

File: testing/features/support/components/search/standard.rb [hunk 2]
```diff
@@ -18,6 +18,7 @@ def search_for(term)
     private
 
     def clear_field
+      search_field.send_keys(%i[control end]) if firefox?
```
Comment (luke-hill) [inline testing/features/support/components/search/standard.rb:21]: Is this something you were debugging? This shouldn't be required on firefox>?

**Review by luke-hill — CHANGES_REQUESTED** [review]
Couple of things to quickly triage.

File: testing/features/support/components/search/standard.rb (hunk 2 above)
Comment (celonicolosi) [reply testing/features/support/components/search/standard.rb:21]: The tests were falling for Firefox, then I realized that just send_keys(:end) was not sending the caret navigation to the end of the text, resulting in the following error:

Screenshot 2021-02-26 at 13 14 57

```
RSpec::Expectations::ExpectationNotMetError: 
expected: "Anything"
     got: "AnythingCurrent search term" 
```

File: testing/features/support/components/header/standard.rb (hunk 1 above)
Comment (celonicolosi) [reply testing/features/support/components/header/standard.rb:11]: **Sign in / Sign out element**
```
<li class="cads-header__account-link">
    <form action="/sign-out/out" class="cads-header__account-form" method="POST">
        <button class="cads-linkbutton__regular">Sign out</button>
    </form>
</li>
```

Expecting for login (element :login, "[data-testid='account-link']")
`expect(@component).to have_login`

Generate the following error:
```
RSpec::Expectations::ExpectationNotMetError: expected #<Header::Standard:0x00007fa2161373f0 @loaded=true, @load_error=nil> to have login
...
/Users/marcelonicolosi/Projects/design-system/testing/features/step_definitions/header_steps.rb:12:in `"a login/logout link is present"'
/Users/marcelonicolosi/Projects/design-system/testing/features/header.feature:26:in `a login link is present'
```

File: testing/features/support/components/search/standard.rb (hunk 2 above)
Comment (luke-hill) [reply testing/features/support/components/search/standard.rb:21]: Right, oki. This shouldn't be the case. See if you can find out what's causing it. I'll dig into it on Monday too if needs be.

This should work for all browsers. If it's not something is likely wrong with our code.

File: testing/features/support/components/header/standard.rb (hunk 1 above)
Comment (luke-hill) [reply testing/features/support/components/header/standard.rb:11]: Looking at the haml it's likely we just need to add the test id into both branches of the if statement.

File: testing/features/support/components/search/standard.rb (hunk 2 above)
Comment (celonicolosi) [reply testing/features/support/components/search/standard.rb]: In the end, I didn't find out why `search_field.send_keys(:end)` wasn't working properly on Firefox.
I just find out that the array of symbols format works fine for Firefox `search_field.send_keys(%i[end])`, unlike the default `search_field.send_keys(:end)`. The array of symbol works fine on chrome too.

File: testing/features/support/components/header/standard.rb (hunk 1 above)
Comment (celonicolosi) [reply testing/features/support/components/header/standard.rb:11]: I didn't fully get it! Add the data-testid again on the front-end?

Comment (luke-hill) [reply testing/features/support/components/header/standard.rb:11]: Yep, I'll give you a call this morning if you need help

File: testing/features/support/components/search/standard.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@ def search_for(term)
     private
 
     def clear_field
-      search_field.send_keys(:end)
+      search_field.send_keys(%i[end])
```
Comment (luke-hill) [inline testing/features/support/components/search/standard.rb:21]: Either we both have or I have seen this error before.

I think going forward we just need to be mindful to always use arrays for this method. I'll raise a minor issue in ruby

**Review by luke-hill — CHANGES_REQUESTED** [review]
Make sure to push up the haml changes too.

**Review by luke-hill — APPROVED** [review]

**Comment by luke-hill** [comment]
@baseonmars Just for sanity can you once over these changes (We edit haml here slightly).

**Review by davidrapson — APPROVED** [review]
👍 HAML changes look good to me

**Pull request merged**